cmake/OpenCVDetectCUDA.cmake

```diff
@@ -28,6 +28,19 @@ endif()
 
 if(CUDA_FOUND)
   set(HAVE_CUDA 1)
+  if(CUDA_VERSION VERSION_GREATER_EQUAL "11.0")
+    # CUDA 11.0 removes nppicom
+    foreach(component IN LISTS CUDA_nppi_LIBRARY)
+      if(component MATCHES "nppicom")
+        list(REMOVE_ITEM CUDA_nppi_LIBRARY ${component})
+      endif()
+    endforeach()
+    foreach(component IN LISTS CUDA_npp_LIBRARY)
+      if(component MATCHES "nppicom")
+        list(REMOVE_ITEM CUDA_npp_LIBRARY ${component})
+      endif()
+    endforeach()
```

Comment:
Please try to use [ocv_list_filterout()](https://github.com/opencv/opencv/blob/4.3.0/cmake/OpenCVUtils.cmake#L1050) macro:
```
ocv_list_filterout(CUDA_nppi_LIBRARY "nppicom")
ocv_list_filterout(CUDA_npp_LIBRARY "nppicom")
```